cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -475,4 +480,17 @@ private NamespaceId getNamespaceId(byte[] rowKey) {
   public interface DeleteCallback {
     void handle(NamespaceId namespaceId, Location location, String namespacedLogBaseDir);
   }
+
+  private static Location getCompatibleLocationFromValue(LocationFactory locationFactory,
+                                                         byte[] value) throws URISyntaxException {
```

Comment:
doesn't have to be static right ? also location factory is available in the instance, doesn't have to pass if its non-static.